src/test/java/wooteco/subway/controller/LineControllerTest.java

```diff
@@ -0,0 +1,143 @@
+package wooteco.subway.controller;
+
+import io.restassured.RestAssured;
+import io.restassured.response.ExtractableResponse;
+import io.restassured.response.Response;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import wooteco.subway.dao.LineDao;
+import wooteco.subway.domain.Line;
+import wooteco.subway.dto.LineResponse;
+import wooteco.subway.exception.NotFoundException;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+class LineControllerTest extends AcceptanceTest {
+
+    @Autowired
+    private LineDao lineDao;
+
+    @DisplayName("지하철 노선을 등록한다.")
+    @Test
+    void createLine() {
+        Map<String, String> params = new HashMap<>();
+        params.put("name", "신분당선");
+        params.put("color", "red");
+
+        ExtractableResponse<Response> response = RestAssured.given().log().all()
+                .body(params)
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .when()
+                .post("/lines")
+                .then().log().all()
+                .extract();
+
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.CREATED.value());
+        assertThat(response.header("Location")).isNotBlank();
+    }
+
+    @DisplayName("중복된 이름을 가진 지하철 노선을 등록할 때 예외를 발생시킨다.")
+    @Test
+    void throwsExceptionWhenCreateDuplicatedName() {
+        lineDao.save(new Line("신분당선", "red"));
+
+        Map<String, String> params = new HashMap<>();
+        params.put("name", "신분당선");
+        params.put("color", "red");
+
+        ExtractableResponse<Response> response = RestAssured.given().log().all()
+                .body(params)
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .when()
+                .post("/lines")
+                .then().log().all()
+                .extract();
+
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.BAD_REQUEST.value());
+    }
+
+    @DisplayName("지하철 노선 목록을 조회한다.")
+    @Test
+    void getLines() {
+        lineDao.save(new Line("신분당선", "red"));
+        lineDao.save(new Line("1호선", "blue"));
+
+        ExtractableResponse<Response> response = RestAssured.given().log().all()
+                .when()
+                .get("/lines")
+                .then().log().all()
+                .extract();
+        List<LineResponse> actual = response.jsonPath().getList(".", LineResponse.class);
+
+        assertThat(actual.size()).isEqualTo(2);
```

Comment:
```suggestion
        assertThat(actual).hasSize(2);
```

이런 메서드도 있어요 😎

그런데, 크기 검증만으로 충분할까요? 내부에서 값을 아무렇게나 바꿔서 리턴해도 크기만 같으면 테스트가 통과하게 되는 것 아닐까요?

Reply:
`assertThat().hasSize()`메서드는 처음 본 것 같네요🥸!!

반환 리스트의 크기만 검증하는 테스트 코드들을 내부 값들을 검증하도록 수정했습니다😁